core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -2503,4 +2504,53 @@ public void testAddRequiredUnknown() {
         .isInstanceOf(IllegalArgumentException.class)
         .hasMessage("Cannot create required field with unknown type: unk");
   }
+
+/**
+   * This test reproduces the issue where dropping a column with the highest field ID
+   * fails with "Invalid last column ID: X < Y (previous last column ID)"
```

Comment:
both tests pass for me locally without applying any of the fixes from this PR

Reply:
thanks, let me check